l10n_ir_base/models/user_extensions.py

```diff
@@ -0,0 +1,13 @@
+from odoo import fields, models
+
+
+class UsersExtensions(models.Model):
+    _inherit = "res.users"
+
+    calendar = fields.Selection(
+        [("gregorian", "Gregorian Calendar"), ("jalali", "Jalali (Persian) Calendar")],
+        default="gregorian",
+    )
+    date_format = fields.Selection(
```

Comment:
What about using directly the user.lang.date_format instead?

I mean, the following field:
 - https://github.com/odoo/odoo/blob/abc9fdaae2927214d98082f391a1dc0fc75e4c77/odoo/addons/base/models/res_lang.py#L34-L35

Reply:
in fact, calendar shouldn't be configured in the lang too?

Reply:
yes, I understand
But the level of technical knowledge of users is low
 And I think the person who installs the localization module should be able to quickly change the calendar in the profile settings.
On the other hand, the date format
In Iran, it is accepted only in the 2 mentioned formats

Of course, one of the people advising changed his mind and emphasized a format
I think
The goal is to localize the convenience of people who are looking to use odoo but with their own country calendar
https://apps.odoo.com/apps/modules/14.0/jalali_calendar/   675.00 €
https://apps.odoo.com/apps/modules/10.0/jalali_cal/     598.00 €

I reported this to the odoo developers
 And soon the calendar selection will be added to odoo
 And this module will change

Reply:
IMHO If you use the lang (records of res.lang are pre-created) you can pre-configure the lang to use this format even without user intervention or extra configuration for each users

I mean, currently the user is using the lang English?

It is using its language and you can set the correct calendar for its

But it is only a recommendation not blocking

thank you for your efforts here

Reply:
Thanks for your help, I will definitely add this item
Persian language may work with the Gregorian calendar, but by default it worked as jalali
The date format can also be used

These items after completing the module "l10n_ir_web"
Will add